File: CarHomeWidget/src/main/java/com/anod/car/home/utils/TrialDialogs.java
package com.anod.car.home.utils;

import com.anod.car.home.R;

import android.support.v7.app.AlertDialog;
import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.content.DialogInterface.OnClickListener;
import android.content.Intent;

public class TrialDialogs {

    public static Dialog buildProInstalledDialog(Context context) {
        AlertDialog.Builder builder = getBuilder(context);
        builder.setTitle(R.string.dialog_donate_title_install);
        builder.setMessage(R.string.dialog_donate_message_installed);
        builder.setPositiveButton(android.R.string.ok, new OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
            }
        });
        return builder.create();
    }


    public static Dialog buildTrialDialog(int trialsLeft, final Context context) {
        AlertDialog.Builder builder = getBuilder(context);
        int negativeRes = 0;
        if (trialsLeft > 0) {
            String activationsLeftText = context.getResources()
                    .getQuantityString(R.plurals.notif_activations_left, trialsLeft, trialsLeft);

            builder.setTitle(R.string.dialog_donate_title_trial);
            builder.setMessage(
                    activationsLeftText + context.getString(R.string.dialog_donate_message_trial));
            negativeRes = R.string.dialog_donate_btn_trial;
        } else {
            builder.setTitle(R.string.dialog_donate_title_expired);
            builder.setMessage(R.string.dialog_donate_message_expired);
            negativeRes = R.string.dialog_donate_btn_no;
        }
        setTrialDialogButtons(context, builder, negativeRes);
        return builder.create();
    }


    public static Dialog buildProOnlyDialog(final Context context) {
        AlertDialog.Builder builder = getBuilder(context);
        builder.setTitle(R.string.app_name);
        builder.setMessage(R.string.dialog_donate_message_expired);
        setTrialDialogButtons(context, builder, R.string.dialog_donate_btn_no);
        return builder.create();
    }

    private static void setTrialDialogButtons(final Context context, AlertDialog.Builder builder,
            int negativeRes) {
        builder.setNeutralButton(R.string.dialog_donate_btn_yes, new OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                Intent intent = IntentUtils.createProVersionIntent();
                Utils.startActivitySafely(intent, context);
                dialog.dismiss();
            }

        });
        builder.setNegativeButton(negativeRes, new OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
            }
        });
    }


    private static AlertDialog.Builder getBuilder(final Context context) {
        AlertDialog.Builder builder = new AlertDialog.Builder(context);
        builder.setCancelable(true);
        return builder;
    }

}

File: app/src/main/java/com/anod/car/home/utils/TrialDialogs.kt
package com.anod.car.home.utils

import android.app.Dialog
import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AlertDialog
import com.anod.car.home.R
import info.anodsplace.carwidget.utils.forProVersion
import info.anodsplace.framework.app.DialogMessage
import info.anodsplace.framework.content.startActivitySafely

object TrialDialogs {

    fun buildTrialDialog(trialsLeft: Int, context: Context): Dialog {
        val message = if (trialsLeft > 0) {
            val activationsLeftText = context.resources
                    .getQuantityString(R.plurals.notif_activations_left, trialsLeft, trialsLeft)
            activationsLeftText + context.getString(R.string.dialog_donate_message_trial)
        } else {
            context.getString(R.string.dialog_donate_message_expired)
        }
        val title = if (trialsLeft > 0) R.string.dialog_donate_title_trial else R.string.dialog_donate_title_expired
        return DialogMessage(
                context,
            R.style.Alert,
                title,
                message
        ) {
            val negativeRes = if  (trialsLeft > 0) R.string.dialog_donate_btn_trial else R.string.dialog_donate_btn_no
            setTrialDialogButtons(context, it, negativeRes)
        }.create()
    }

    private fun setTrialDialogButtons(context: Context, builder: AlertDialog.Builder,
                                      negativeRes: Int) {
        builder.setNeutralButton(R.string.dialog_donate_btn_yes) { dialog, _ ->
            val intent = Intent().forProVersion()
            context.startActivitySafely(intent)
            dialog.dismiss()
        }
        builder.setNegativeButton(negativeRes) { dialog, _ -> dialog.dismiss() }
    }

}